gmplot/gmplot.py

```diff
@@ -325,20 +327,32 @@ def write_map(self,  f):
         f.write('\t\t};\n')
         f.write(
             '\t\tvar map = new google.maps.Map(document.getElementById("map_canvas"), myOptions);\n')
+        f.write('\t\tvar markers = [];\n')
         f.write('\n')
 
-    def write_point(self, f, lat, lon, color, title):
+    def write_point(self, f, lat, lon, color, title, info_window_html=""):
         f.write('\t\tvar latlng = new google.maps.LatLng(%f, %f);\n' %
                 (lat, lon))
         f.write('\t\tvar img = new google.maps.MarkerImage(\'%s\');\n' %
                 (self.coloricon % color))
-        f.write('\t\tvar marker = new google.maps.Marker({\n')
-        f.write('\t\ttitle: "%s",\n' % title)
-        f.write('\t\ticon: img,\n')
-        f.write('\t\tposition: latlng\n')
-        f.write('\t\t});\n')
-        f.write('\t\tmarker.setMap(map);\n')
+        f.write('\t\tmarkers.push(new google.maps.Marker({\n')
+        f.write('\t\t\ttitle: "%s",\n' % title)
+        f.write('\t\t\ticon: img,\n')
+        f.write('\t\t\tposition: latlng\n')
+        f.write('\t\t}));\n')
+        if info_window_html != "":
+            f.write("\t\tgoogle.maps.event.addListener(markers[%d], 'click', function () {\n" % self.numMarkers)
+            f.write("\t\t\tinfo_window = new google.maps.InfoWindow({\n")
+            # Escape every single quote and every newline in info_window_html.
+            info_window_html = info_window_html.replace("'", "\\'").replace("\n", "\\n")
+            print("info_window_html transformed to ", info_window_html)
+            f.write("\t\t\t\tcontent: '%s',\n" % info_window_html)
+            f.write("\t\t\t});\n")
+            f.write("\t\t\tinfo_window.open(map, markers[%d]);\n" % self.numMarkers)
```

Comment:
Actually, scratch that - we shouldn't be creating a new `InfoWindow` inside the `onClick` handler, otherwise each click will create a new Info Window on top of the previous one. We'll have to create this once, give it a unique name, then reference it in the `onClick` handler.